src/include/migraphx/process.hpp

```diff
@@ -50,8 +52,14 @@ struct MIGRAPHX_EXPORT process
 
     process& cwd(const fs::path& p);
 
-    void exec();
-    void write(std::function<void(process::writer)> pipe_in);
+    void exec(std::string_view args = "", std::string_view envs = "");
```

Comment:
`args` should be passed to the constructor, and it should be `std::vector<std::string>`. Also, there should be an `env` method similar to `cwd` to set the environment.